### PR TITLE
Static dispatch Arm-optimized-routines instead of dynamic dispatch

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #7193

Problem Summary:

### What is changed and how it works?

hook syscall may make `getauxval` failed which may cause core dump.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: libs/libarm-optimized-routines/aor.h [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+#elif __ARM_NEON
+    return __memcpy_aarch64_simd(dst_, src_, size);
+#endif
+    return __memcpy_aarch64(dst_, src_, size);
+#elif __arm__
+    return __memcpy_arm(dst_, src_, size);
+#else
+    return memcpy(dst_, src_, size);
+#endif
+}
```
Comment (JaySon-Huang) [inline libs/libarm-optimized-routines/aor.h:40]: Can this file get the correct macro for `__ARM_FEATURE_SVE` / `__ARM_NEON`, etc?

Comment (JaySon-Huang) [reply libs/libarm-optimized-routines/aor.h:40]: I'm not sure where these macro is defined. Are they defined in the arm system header?

Comment (Lloyd-Pottiger) [reply libs/libarm-optimized-routines/aor.h:40]: Yes, I have verified.

image


They are introduced by flag, like "+simd" will add macro `__ARM_NEON`.

https://developer.arm.com/documentation/dui0774/l/Other-Compiler-specific-Features/Predefined-macros

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**